src/drv_oculus_rift/rift.c

```diff
@@ -315,6 +315,30 @@ static ohmd_device* open_device(ohmd_driver* driver, ohmd_device_desc* desc)
 	priv->base.properties.lens_vpos = priv->display_info.v_center;
 	priv->base.properties.ratio = ((float)priv->display_info.h_resolution / (float)priv->display_info.v_resolution) / 2.0f;
 
+	//setup generic distortion coeffs, from hand-calibration
+	switch (desc->revision) {
```

Comment:
What would that function do?  Are you trying to abstract it out of the device specific driver code?  i.e. inside the driver you call setUniversalDistoriton(a,b,c,d) instead of modifying the base device struct directly?